WalletWasabi.Gui/Global.cs

```diff
@@ -76,7 +77,7 @@ public class Global
 
 		public JsonRpcServer? RpcServer { get; private set; }
 
-		public Global(string dataDir, Config config, UiConfig uiConfig, WalletManager walletManager)
+		public Global(string dataDir, Config config, UiConfig uiConfig, WalletManager walletManager, CoinJoinManager coinJoinManager)
```

Comment:
Are we sure we want to send `CoinJoinManager` through ctor only to pass it to `SystemAwakeChecker`?
We could register `CoinJoinManager` before we register `SystemAwakeChecker` and just grab the `CJManager` with `HostedServices.GetOrDefault`.

Reply:
Objects should always be transparent as possible regarding their dependencies.